src/Migration/Sources/CSV.php

```diff
@@ -0,0 +1,281 @@
+<?php
+
+namespace Utopia\Migration\Sources;
+
+use Utopia\Database\Database as UtopiaDatabase;
+use Utopia\Migration\Exception;
+use Utopia\Migration\Resource;
+use Utopia\Migration\Resources\Database\Attribute;
+use Utopia\Migration\Resources\Database\Collection;
+use Utopia\Migration\Resources\Database\Database;
+use Utopia\Migration\Resources\Database\Document;
+use Utopia\Migration\Resources\Storage\File;
+use Utopia\Migration\Source;
+use Utopia\Migration\Sources\Appwrite\Reader;
+use Utopia\Migration\Sources\Appwrite\Reader\Database as DatabaseReader;
+use Utopia\Migration\Transfer;
+use Utopia\Storage\Device;
+
+class CSV extends Source
+{
+    private string $filePath;
+
+    /**
+     * format: `{databaseId:collectionId}`
+     */
+    private string $resourceId;
+
+    private Device $device;
+
+    private Reader $database;
+
+    public function __construct(
+        string $resourceId,
+        string $filePath,
+        Device $device,
+        ?UtopiaDatabase $dbForProject
+    ) {
+        $this->device = $device;
+        $this->filePath = $filePath;
+        $this->resourceId = $resourceId;
+        $this->database = new DatabaseReader($dbForProject);
+    }
+
+    public static function getName(): string
+    {
+        return 'CSV';
+    }
+
+    public static function getSupportedResources(): array
+    {
+        return [
+            Resource::TYPE_DOCUMENT,
+        ];
+    }
+
+    // called before the `exportGroupDatabases`.
+    public function report(array $resources = []): array
+    {
+        $report = [];
+
+        $this->withCsvStream(function ($stream) use (&$report) {
+            $headers = fgetcsv($stream);
+            if (! is_array($headers) || count($headers) === 0) {
+                return;
+            }
+
+            $rowCount = 0;
+            while (fgetcsv($stream) !== false) {
+                $rowCount++;
+            }
+
+            $report[Resource::TYPE_DOCUMENT] = $rowCount;
+        });
+
+        return $report;
```

Comment:
Can you see if there's a more efficient way to do this? Maybe with `SplFileObject`?

Reply:
Done.